src/test/java/com/example/project/test/AETest.java

```diff
@@ -0,0 +1,62 @@
+package com.example.project.test;
+
+import com.example.project.page_object.*;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Test;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class AETest {
+
+    private WebDriver driver;
+
+    @BeforeSuite
+    public void setUp() {
+        driver = new ChromeDriver();
+        driver.get("https://www.ae.com/us/en");
+        driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
+    }
+
+    @Test
+    public void buyProductTest() throws InterruptedException {
+        String style = "1399-5795";
+        String color = "300";
+
+        MainPage mainPage = new MainPage(driver);
+        mainPage.acceptCookies();
+        mainPage.searchForProduct(style + " " + color);
+
+        SearchResultsPage searchResultsPage = new SearchResultsPage(driver);
+        searchResultsPage.clickOnProductTile();
+
+        ProductPage productPage = new ProductPage(driver);
+        assertThat(productPage.style.getText()).isEqualTo(style);
```

Comment:
these assertions do not seem to make much sense in current implementation. As style and colour locators are tied to the test they contain, so if the text does not match they simply would not be found.